SAR-160 - Readmission measure

anthem123 opened the pull request:
# Related Tickets

- [SAR-160](https://jira.amida-tech.com/browse/SAR-160)

This CQL is used to calculate the readmission HEDIS score by checking if a patient had 2 or more unplanned trips to the hospital where they had to stay for more than 1 day within 30 days of each other.

The full list of exceptions is

- Pregnancy and perinatal visits
- Chemotherapy
- Kidney transplant
- Bone marrow transplant
- Organ transplant
- Introduction of pancreatic cells

The final score is

Total readmissions within 30 days / Total applicable admissions

anthem123 committed:
Created initial CQL and data files.

anthem123 committed:
Added numerator calculations.

anthem123 committed:
Added missing data to run CQL file, fixed direct transfer checks to return correct data type.

anthem123 committed:
Started risk adjusted weight calculations.

anthem123 committed:
Denominator now uses the ExplanationOfBenefit to link coverage with encounters.

anthem123 committed:
Removed SES Stratification calculations. Closer to calculating numerator score.

anthem123 committed:
Finalized numerator and denominator calculations.

anthem123 committed:
Added more comments, borrowed some logic from SAR-159

eawest2 commented:
I don't get the usual logs showing the results of the CQL in the console. Is this intended?

anthem123 committed:
Updated data sets to check more conditions, fixed code to more accurately check those conditions.

anthem123 committed:
Finished denominator calculations

Each step now filters out one encounter from the list.

anthem123 committed:
Finalized Numerator Calculation

Added valueset for potentially planned procedures.

anthem123 committed:
Fixed value when numerator is 0

If the numerator is zero it was calculating to undefined. Will now calculate to zero.

Also added better data for patient 2 and combined all data into readmission.json

eawest2 reviewed (approved):
LGTM

jmfadeley commented on cql/readmission.cql:
```diff
@@ -0,0 +1,172 @@
+  end
+
+define function hasPregnancyEncounter(eob ExplanationOfBenefit):
+  Exists(([Encounter: "Inpatient Obstetrics NDNQI"] enc where Substring(eob.item[0].encounter[0].reference.value, 9) = enc.id))
```
Check if we could add the Perinatal part with https://vsac.nlm.nih.gov/valueset/2.16.840.1.113762.1.4.1195.334/expansion/Latest

jmfadeley commented on cql/readmission.cql:
```diff
@@ -0,0 +1,172 @@
+    when 'Medicare' then if Count("Continuous Enrollment") < 4 then "Continuous Enrollment" else List<ExplanationOfBenefit>{}
+    when 'Medicaid' then if Count("Continuous Enrollment") < 4 then "Continuous Enrollment" else List<ExplanationOfBenefit>{}
+    else if Count("Continuous Enrollment") < 3 then "Continuous Enrollment" else List<ExplanationOfBenefit>{}
+  end
```
Let's create a new ticket to investigate if we need to report a separate outlier ticket. See Step 6 for explanation. Make it a 2 point evaluation ticket and expand to 5 points to solve if needed.

jmfadeley commented on cql/readmission.cql:
```diff
@@ -0,0 +1,172 @@
+    union [Encounter: "Kidney Transplant"]
+    union [Encounter: "Bone Marrow Transplant"]
+    union [Encounter: "Organ Transplant"]
+    union [Encounter: "Inpatient Obstetrics NDNQI"]
```
Don't forget to add the perinatal here.

anthem123 replied:
Perinatal has been added, you have already created pancreatic cell transplant ticket.

jmfadeley commented on cql/readmission.cql:
```diff
@@ -0,0 +1,172 @@
+    union [Encounter: "Bone Marrow Transplant"]
+    union [Encounter: "Organ Transplant"]
+    union [Encounter: "Inpatient Obstetrics NDNQI"]
+    union [Encounter: "Potentially Planned Procedures"]
```
Make this its own function and check for the "Acute Condition Value Set" to exclude that from the PPP grouping above.

anthem123 replied:
Acute Condition Value Set has been added.

anthem123 committed:
Valueset update

Added codes for Acute Conditions and Perinatal Conditions
